android/src/main/java/ly/count/android/sdk/react/CountlyReactNative.java

```diff
@@ -803,6 +825,29 @@ public void endEvent(ReadableArray args) {
         }
     }
 
+    @ReactMethod
+    public void endEvent(ReadableMap args) {
+        String eventName = args.getString("n");
+
+        int eventCount = 0;
+        if (args.hasKey("c")) {
+            eventCount = args.getInt("c");
+        }
+
+        double eventSum = 0;
+        if (args.hasKey("s")) {
+            eventSum = args.getDouble("s");
+        }
+
+        HashMap<String, Object> segmentation = new HashMap<>();
+        ReadableArray segments = args.getArray("g");
+        for (int i = 0, il = segments != null ? segments.size() : 0; i < il; i += 2) {
+            segmentation.put(segments.getString(i), segments.getString(i + 1));
```

Comment:
same here, I guess it is same for all segmentations. While we are at it please add type checks

Reply:
done